### ISSUE TITLE
Enhance: Add yaml backup and restore for application

### ISSUE THREAD

**Issue by jerryhu1234** [kubesphere/kubesphere#3134]
Scenes:
At present, many projects include front-end and back-end micro-services. It happens that kk 3.0 supports applications similar to argocd, which can abstract and manage these resources on deployment, svc, statefulset etc.
demand:
Is it possible to provide the backup or download function (yaml arrangement) of all resources under the application? If it is accidentally deleted, all the microservices under the project will be deleted. This is a big risk ,can provide one-key recovery function?

**Comment by zackzhangkai** [comment]
Firstly, get your application 

```
kubectl get application -n xxx
...
bookinfo  
```

Secondly, execute the command as below:

```
for i in `kubectl get app bookinfo -oyaml | grep -i link | awk '{print $(NF)}'`;do kubectl get --raw ''$i'' >> sample-bookinfo.json.20201123.bak; done  
```

Then you will get all the referenced resources aboud the appliction, the format is json.

**Comment by zryfish** [comment]
> Scenes:
> At present, many projects include front-end and back-end micro-services. It happens that kk 3.0 supports applications similar to argocd, which can abstract and manage these resources on deployment, svc, statefulset etc.
> demand:
> Is it possible to provide the backup or download function (yaml arrangement) of all resources under the application? If it is accidentally deleted, all the microservices under the project will be deleted. This is a big risk ,can provide one-key recovery function?

Yes, that's a popular feature request. We are working on a solution to backup and restore objects for KubeSphere, stay tuned.
